src/sac_stac/domain/s3.py

```diff
@@ -89,11 +89,12 @@ def list_common_prefixes(self, bucket_name, prefix):
             prefix                 (str): Prefix
         """
         common_prefixes = []
-        paginator = self.s3_resource.meta.client.get_paginator('list_objects')
-
-        for result in paginator.paginate(Bucket=bucket_name, Prefix=prefix, Delimiter='/'):
-            if result.get('CommonPrefixes'):
-                common_prefixes = [p.get('Prefix') for p in result.get('CommonPrefixes')]
+        paginator = self.s3_resource.meta.client.get_paginator('list_objects_v2')
+        pages = paginator.paginate(Bucket=bucket_name, Prefix=prefix, Delimiter='/')
+        for page in pages:
+            if page.get('CommonPrefixes'):
+                for p in page.get('CommonPrefixes'):
```

Comment:
could use common_prefixes.extend here rather than looping again.

Reply:
thanks! could you show an example of this? just how to get the extra p.get('Prefix') on each iteration without looping again?

Reply:
humm, actually because you need to call get('Prefix') on each one it doens't make much difference.

Best I get ends up like this which is functionally the same...

common_prefixes.extend([p.get('Prefix') for p in page.get('CommonPrefixes')])